python/tvm/module.py

```diff
@@ -118,21 +122,22 @@ def export_library(self,
             self.save(file_name)
             return
 
-        if not (self.type_key == "llvm" or self.type_key == "c"):
-            raise ValueError("Module[%s]: Only llvm and c support export shared" % self.type_key)
+        module = self._lookup_dso_module()
+        assert module is not None, "We should have one dso module to export"
```

Comment:
Is it a hard requirement? things should also works fine if all modules are non dso?